### DISCUSSION TITLE
Configuration option for prerendering pages (overrides output mode)

### DISCUSSION THREAD

**Discussion by humanfriend22** [withastro/roadmap#618] [Proposal]
### Body

# Summary

A configuration option to specify which files to prerender when using the `hybrid` or `server` output mode. We could allow `build.prerenderPages` to take a:

boolean -> to globally prerender all pages while maintaining server API routes, etc
array -> an array of paths to hard code files to prerender
function -> custom logic to decide whether to prerender or not

This is probably the opposite of the hybrid mode.

# Background & Motivation

I needed all my pages to be pre-rendered but had some backend apis I wanted to be maintained.

# Goals

- An additional configuration option (preferably `build.prerenderPages`)

# Example

This would be opt-in and would not break any existing code as developers can either add `build.prerenderPages` to their project or `export const prerender = boolean`.

# PR?

This is my first real project with Astro and don't have a great understanding of the API so this might already exist but I found out [where](https://github.com/withastro/astro/blob/8ee94ffbd9b0c055a245f63e766f64f51466eda7/packages/astro/src/prerender/metadata.ts#L21) `export const prerender = boolean` is being read and it doesn't seem it is being influenced by the config.

**Comment by florian-lefebvre** [comment]
I think this is possible already: you have the global `output` which can be overriden on a page basis. Otherwise, you can use this integration hook: https://docs.astro.build/en/reference/integrations-reference/#astroroutesetup
